src/transformers/trainer.py

```diff
@@ -417,6 +417,42 @@ def __init__(
             elif FSDPOption.NO_SHARD in args.fsdp:
                 self.fsdp = ShardingStrategy.NO_SHARD
 
+        if args.xla_fsdp:
+            try:
+                from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as FSDP, checkpoint_module
+            except ImportError:
+                assert False, "Missing module XLAFullyShardedDataParallel; this module is available in torch-xla >= 1.12.0."
+            fsdp_kwargs = args.xla_fsdp_config
+            fsdp_wrap = lambda m: FSDP(m, **fsdp_kwargs)
+            # A wrapper for gradient checkpointing
+            grad_ckpt_wrap = checkpoint_module if args.xla_fsdp_grad_ckpt else (lambda m: m)
+            if args.xla_fsdp_nested:
+                if not(hasattr(model, 'transformer') and hasattr(model.transformer, 'h')):
+                    raise ValueError(
+                        "Nested XLA FSDP is currently only supported for models which expose their"
+                        " transformer blocks through `transformer.h`."
+                    )
```

Comment:
That's just GPT/GPT-2. This is too restrictive to be merged like this in the Trainer.